erpnext/stock/doctype/warehouse/warehouse.py

```diff
@@ -138,6 +139,15 @@ def convert_to_group(self):
 			self.save()
 			return 1
 
+	def unlink_from_items(self):
+		item_default_names = frappe.db.get_list(
+			'Item Default',
+			pluck='name',
+			filters={'default_warehouse':self.name}
+		)
+		for name in item_default_names:
+			frappe.get_doc("Item Default", name).delete()
```

Comment:
Use `frappe.delete_doc` instead of get_doc().delete()? `get_doc()` makes additional db calls to fetch data (that's about to be nuked anyway).

https://frappeframework.com/docs/user/en/api/document#frappedelete_doc